ikarus/controlroutines/pathfollowingtechnique.hh

```diff
@@ -16,54 +18,75 @@
 
 namespace Ikarus {
 
-  struct PathFollowingInformation {
-    bool success{false};
-  };
-
-  template <typename NonLinearSolver, typename PathFollowingType = Ikarus::StandardArcLength>
-  requires Concepts::PathFollowingStrategy<PathFollowingType, typename NonLinearSolver::NonLinearOperator>
-  class PathFollowing : public IObservable<ControlMessages> {
+  template <typename NonLinearSolver, typename PathFollowingType = Ikarus::StandardArcLength,
+            typename AdaptiveStepSizing = Ikarus::NoAdaptiveStepSizing<>>
+  requires((Concepts::PathFollowingStrategy<
+               PathFollowingType, typename NonLinearSolver::NonLinearOperator,
+               Ikarus::SubsidiaryArgs>)and(Concepts::
+                                               AdaptiveStepSizingStrategy<
+                                                   AdaptiveStepSizing, Ikarus::NonLinearSolverInformation,
+                                                   Ikarus::SubsidiaryArgs,
+                                                   std::remove_cvref_t<typename NonLinearSolver::NonLinearOperator>>)
+           and (Concepts::NonLinearSolverCheckForPathFollowing<NonLinearSolver>)) class PathFollowing
+      : public IObservable<Ikarus::ControlMessages> {
   public:
     PathFollowing(const std::shared_ptr<NonLinearSolver>& p_nonLinearSolver, int loadSteps, double stepSize,
-                  PathFollowingType p_pathFollowingType = Ikarus::StandardArcLength{})
+                  PathFollowingType p_pathFollowingType   = Ikarus::StandardArcLength{},
+                  AdaptiveStepSizing p_adaptiveStepSizing = {})
         : nonLinearSolver{p_nonLinearSolver},
           loadSteps_{loadSteps},
           stepSize_{stepSize},
-          pathFollowingType_{p_pathFollowingType} {}
+          pathFollowingType_{p_pathFollowingType},
+          adaptiveStepSizing{p_adaptiveStepSizing} {}
 
-    PathFollowingInformation run() {
-      spdlog::info("Started path following with subsidiary equation: {}", pathFollowingType_.name);
-      PathFollowingInformation info({false});
+    ControlInformation run() {
+      ControlInformation info({false});
       auto& nonOp = nonLinearSolver->nonLinearOperator();
-      this->notify(ControlMessages::CONTROL_STARTED);
+      this->notify(ControlMessages::CONTROL_STARTED, pathFollowingType_.name);
+
+      auto start = std::chrono::high_resolution_clock::now();
```

Comment:
This timing should happen in some observer. 
There you can also call [start and stop](std::chrono::high_resolution_clock::now();
Every time you get notified with `CONTROL_STARTED` and `CONTROL_ENDED`.